### PR TITLE
Add queue service protocol layer

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-queues/NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,483 @@
+azure-storage-blobs
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-queues/NOTICE.txt:1]: queues

Comment (Jinming-Hu) [inline sdk/storage/azure-storage-queues/NOTICE.txt:1]:
```suggestion
azure-storage-queues
```

File: sdk/storage/azure-storage-queues/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Release History
+
+## 12.0.0-beta.1 (Unreleased)
+
+### New Features
+
+- Nothing yet.
```
Comment (antkmsft) [inline sdk/storage/azure-storage-queues/CHANGELOG.md:7]:
```suggestion
## 12.0.0-beta.1 (Unreleased)

- Initial release
```

suggestion; This is the phrasing we used for Core's initial release.

File: sdk/storage/azure-storage-queues/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+include(AzureTransportAdapters)
+include(AzureDoxygen)
+include(AzureGlobalCompileOptions)
+
+az_vcpkg_integrate()
+
+if(NOT AZ_ALL_LIBRARIES)
+  find_package(azure-storage-common-cpp CONFIG QUIET)
+  if(NOT azure-storage-common-cpp_FOUND)
+    find_package(azure-storage-common-cpp REQUIRED)
```
Comment (antkmsft) [inline sdk/storage/azure-storage-queues/CMakeLists.txt:25]:
```suggestion
  find_package(azure-storage-common-cpp "12.0.0" CONFIG QUIET)
  if(NOT azure-storage-common-cpp_FOUND)
    find_package(azure-storage-common-cpp "12.0.0" REQUIRED)
```
(this was added just a day ago, and this is how we should do it now. If Queues require updates to Storage-Common and would require greater version, update it here)

File: sdk/storage/azure-storage-queues/vcpkg/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "description": [
+    "Microsoft Azure Storage Queues SDK for C++",
+    "This library provides Azure Storage Queues SDK."
+  ],
+  "homepage": "https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/storage/azure-storage-queues",
+  "license": "MIT",
+  "dependencies": [
+    {
+      "name": "azure-storage-common-cpp",
+      "default-features": false
```
Comment (antkmsft) [inline sdk/storage/azure-storage-queues/vcpkg/vcpkg.json:16]:
```suggestion
      "default-features": false,
      "version>=": "12.0.0"
```
(this was added just a day ago, and this is how we should do it now. If Queues require updates to Storage-Common and would require greater version, update it here)

**Review by antkmsft — APPROVED** [review]
Thank you for picking up all the changes, I think this is a very solid starting point, and you covered everything. Just one thing, we added version specifications just recently (PR went in yesterday?), so I suggested the changes.

File: sdk/storage/azure-storage-queues/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Release History
+
+## 12.0.0-beta.1 (Unreleased)
+
+### New Features
```
Comment (antkmsft) [inline sdk/storage/azure-storage-queues/CHANGELOG.md:5]:
```suggestion
### Features Added
```
Per https://github.com/Azure/azure-sdk-for-cpp/pull/2527#discussion_r662516141, this is how we are going to name sections from now on.

File: sdk/storage/azure-storage-queues/CHANGELOG.md (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-queues/CHANGELOG.md:7]: It's going to be rewritten anyway when we ship the first release. But fine, let me commit your suggestion,

File: sdk/storage/azure-storage-queues/CHANGELOG.md (hunk 5 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-queues/CHANGELOG.md]: I noticed PR https://github.com/Azure/azure-sdk-for-cpp/pull/2534#issuecomment-872494646 was closed. Do you still want to use the new headings? If so, the headings in the changelog wouldn't be consistent.

Comment (antkmsft) [reply sdk/storage/azure-storage-queues/CHANGELOG.md]: That PR was about applying this for past releases. New releases go with the new headings.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-queues/CHANGELOG.md]: all right

**Pull request merged**